pandas/core/frame.py

```diff
@@ -4205,6 +4205,16 @@ def __setitem__(self, key, value) -> None:
                 warnings.warn(
                     _chained_assignment_msg, ChainedAssignmentError, stacklevel=2
                 )
+        # elif not PYPY and not using_copy_on_write():
+        elif not PYPY and warn_copy_on_write():
+            if sys.getrefcount(self) <= 3:  # and (
+                #     warn_copy_on_write()
+                #     or (
+                #         not warn_copy_on_write()
+                #         and self._mgr.blocks[0].refs.has_reference()
+                #     )
+                # ):
+                warnings.warn("ChainedAssignmentError", FutureWarning, stacklevel=2)
```

Comment:
Ẁe don't seem to have tests for where this would warn in the default mode (if you set a column here, like `df[slice]["col"] = ..` that already nevers update the original df, and you already get a (somewhat incorrect?, as you are not actually "setting on a  copy", you are just not setting inplace) SettingWithCopyWarning.

But to fabricate something that we should warn about in the default mode as well:

```
In [1]: df = DataFrame({"a": [1, 2, 3, 4, 5], "b": 1})

In [2]: df[0:3][0:2] = 10
<ipython-input-2-0c4324ecc90c>:1: SettingWithCopyWarning: 
...
```

This updates the original `df`.

So will have to update the above for that.

Reply:
I started on this locally, but given the many other open PRs that will give conficts, maybe I can leave finishing `DataFrame.__setitem__` proper warnings for default mode for a follow-up PR.